### PR TITLE
[Al Amin Step 20] Labels added to task and made it searchable

### PR DESCRIPTION
**Pull request by rr-mdalauddin-alamin**
### Step 20: Let's be able to put labels on tasks

- Let's allow tasks to have multiple labels
- Let's make it possible to search by the attached label

### CODE REVIEW

File: myapp/app/controllers/admin/labels_controller.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+  def edit
+    @label = Label.find(params[:id])
+  end
+
+  def create
+    @label = Label.new(label_params)
+    if @label.save
+      flash[:success] = t('label.flashes.success.created')
+      redirect_to admin_labels_path
```
Comment (stakahashi-d) [inline myapp/app/controllers/admin/labels_controller.rb]: you can write in one-liner

``` rb
redirect_to admin_labels_path, flash: { success:  t('label.flashes.success.created')}
```

File: myapp/app/controllers/admin/labels_controller.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      flash[:notice] = t('label.flashes.success.updated')
+      redirect_to admin_labels_path
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    @label = Label.find(params[:id])
+    puts(@label.tasks)
```
Comment (stakahashi-d) [inline myapp/app/controllers/admin/labels_controller.rb]: forget to remove?

File: myapp/app/controllers/admin/labels_controller.rb [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    @label.tasks.destroy_all
```
Comment (stakahashi-d) [inline myapp/app/controllers/admin/labels_controller.rb:36]: Is it OK to remove all the tasks related to the label?
I just thought that it was a little bit weird

Comment (stakahashi-d) [reply myapp/app/controllers/admin/labels_controller.rb:36]: When 2 labels were set to the task, and 1 was deleted, that task will be also deleted...

**Comment by stakahashi-d** [comment]
It is ok for this PR, but let's add some screenshots or video, so that the reviewer can know what you expected, or what you tested.
https://github.com/Fablic/training/pull/1489

File: myapp/app/controllers/admin/labels_controller.rb (hunk 1 above)
Comment (rr-mdalauddin-alamin) [reply myapp/app/controllers/admin/labels_controller.rb]: modified

File: myapp/app/controllers/admin/labels_controller.rb (hunk 2 above)
Comment (rr-mdalauddin-alamin) [reply myapp/app/controllers/admin/labels_controller.rb]: yeah :D

Comment (rr-mdalauddin-alamin) [reply myapp/app/controllers/admin/labels_controller.rb]: removed

File: myapp/app/controllers/admin/labels_controller.rb (hunk 3 above)
Comment (rr-mdalauddin-alamin) [reply myapp/app/controllers/admin/labels_controller.rb:36]: umm... isn't it about the pivot table entry removing?

suppose we have 2 labels attached to a single task. and the entry's are in `tasks_labels` joint table. as `{task_id => 1, label_id = 1, task_id => 1, label_id = 2}` so removing `{label_id = 2}` will only remove from `tasks_labels` not from `tasks` table

Comment (stakahashi-d) [reply myapp/app/controllers/admin/labels_controller.rb:36]: Oh, I see, thanks for explaining!

**Review by stakahashi-d — APPROVED** [review]
LGTM!

**Pull request merged**